pkg/backup/service_account_action.go

```diff
@@ -25,28 +25,41 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/sets"
-	rbacclient "k8s.io/client-go/kubernetes/typed/rbac/v1"
 
 	"github.com/heptio/ark/pkg/apis/ark/v1"
+	arkdiscovery "github.com/heptio/ark/pkg/discovery"
 	"github.com/heptio/ark/pkg/kuberesource"
 )
 
 // serviceAccountAction implements ItemAction.
 type serviceAccountAction struct {
 	log                 logrus.FieldLogger
-	clusterRoleBindings []rbac.ClusterRoleBinding
+	clusterRoleBindings []ClusterRoleBinding
 }
 
 // NewServiceAccountAction creates a new ItemAction for service accounts.
-func NewServiceAccountAction(log logrus.FieldLogger, client rbacclient.ClusterRoleBindingInterface) (ItemAction, error) {
-	clusterRoleBindings, err := client.List(metav1.ListOptions{})
+func NewServiceAccountAction(log logrus.FieldLogger, clusterRoleBindingListers map[string]ClusterRoleBindingLister, discoveryHelper arkdiscovery.Helper) (ItemAction, error) {
```

Comment:
It would be nice to have a unit test on this now that it doesn't depend a full clientset.  You could fake/mock a couple of `ClusterRoleBindingListers` to return a specific list of CRBs each, put them in the `clusterRoleBindingListers` arg, then set up a fake `discoveryHelper` with the relevant API groups that correspond to what's in the map (easiest way to set up the fake `discoveryHelper` is probably by not using the constructor function, but just instantiating directly and setting the `apiGroups` to what you need).  Then you can verify that the service account action's `clusterRoleBindings` match what the expected lister had.